api4/config_test.go

```diff
@@ -31,12 +31,12 @@ func TestGetConfig(t *testing.T) {
 		if *cfg.LdapSettings.BindPassword != model.FAKE_SETTING && len(*cfg.LdapSettings.BindPassword) != 0 {
 			require.FailNow(t, "did not sanitize properly")
 		}
-		require.Equal(t, model.FAKE_SETTING, *cfg.FileSettings.PublicLinkSalt, "did not sanitize properly")
+		require.True(t, cfg.FileSettings.PublicLinkSalt == nil || *cfg.FileSettings.PublicLinkSalt == model.FAKE_SETTING, "did not sanitize properly")
 
-		if *cfg.FileSettings.AmazonS3SecretAccessKey != model.FAKE_SETTING && len(*cfg.FileSettings.AmazonS3SecretAccessKey) != 0 {
+		if cfg.FileSettings.AmazonS3AccessKeyId != nil && *cfg.FileSettings.AmazonS3SecretAccessKey != model.FAKE_SETTING && len(*cfg.FileSettings.AmazonS3SecretAccessKey) != 0 {
 			require.FailNow(t, "did not sanitize properly")
 		}
-		if *cfg.EmailSettings.SMTPPassword != model.FAKE_SETTING && len(*cfg.EmailSettings.SMTPPassword) != 0 {
+		if cfg.EmailSettings.SMTPPassword != nil && *cfg.EmailSettings.SMTPPassword != model.FAKE_SETTING && len(*cfg.EmailSettings.SMTPPassword) != 0 {
```

Comment:
@mgdelacroix Is it going to be a problem for local mode of mmctl if it's not able to read config settings without some permissions now?